### ISSUE TITLE
Failure to locate libraries in new manylinux2014 image

### ISSUE THREAD

**Issue by mathstuf** [pypa/manylinux#1310]
Our CI was working with `quay.io/pypa/manylinux2014_x86_64@sha256:9b463efac479efbcab6dec77eca28c5cfa0c5ef64f13ac184eb7117dc1f8edda` but fails with `quay.io/pypa/manylinux2014_x86_64@sha256:1d76fd1dbfa80f5db23faa375fc297a7f9d8bbe11c44d4bbbe562603c47ad0ca`. Taking a failing codebase and running it under the previous image does succeed. The error with the new one is:

```console
$ auditwheel repair dist/*.whl >> $GIT_CLONE_PATH/wheel_output.log
INFO:auditwheel.main_repair:Repairing vtk-9.1.20220421.dev0+mr70228-cp310-cp310-linux_x86_64.whl
Traceback (most recent call last):
  File "/usr/local/bin/auditwheel", line 8, in <module>
    sys.exit(main())
  File "/opt/_internal/pipx/venvs/auditwheel/lib/python3.9/site-packages/auditwheel/main.py", line 59, in main
    rval = args.func(args, p)
  File "/opt/_internal/pipx/venvs/auditwheel/lib/python3.9/site-packages/auditwheel/main_repair.py", line 161, in execute
    out_wheel = repair_wheel(
  File "/opt/_internal/pipx/venvs/auditwheel/lib/python3.9/site-packages/auditwheel/repair.py", line 74, in repair_wheel
    raise ValueError(
ValueError: Cannot repair wheel, because required library "libvtkmdiympi_nompi.so" could not be located
```

The actual library complained about changes from run to run. I suspect something different in the library resolution or `auditwheel` itself, but I haven't dug into the difference between these images yet.

**Comment by mayeut** [comment]
Given cp36 is passing in https://gitlab.kitware.com/vtk/vtk/-/pipelines/277995 and it uses the same image, the diff seems to point to the setuptools update in https://github.com/pypa/manylinux/commit/229b8f9ab22f93bada9da576e400ea72e89bf7a4.
Can you try using latest but downgrade setuptools to check wether or not this is indeed this update that causes issue ?

**Comment by mayeut** [comment]
provided guidance for debugging the issue, no feedback => closing.
please re-open if needed.

**Comment by mathstuf** [comment]
Finally back at this. So I tried in the latest version of the image and it fails. I tried downgrading `setuptools` to 61.3.1, but `auditwheel repair` seems to not care. However, I did a from-scratch build after downgrading and it works. The difference is that the subdirectory under `build` has been changed which disrupts our attempts at appeasing `bdist_wheel` without making our build run from `setup.py` (because it is *far* too complicated to try and re-encode).

FWIW, it looks like 1c23f5e1e4b18b50081cbabb2dea22bf345f5894 is the cause here. It looks like the replacement is to use something like:

```python
from distutils.util import Distribution
from distutils.command import build

d = Distribution()
b = build.build(d)
b.finalize_options()
print(b.build_platlib)
```

Of course `distutils` is now slated for deprecation, so some other way is going to be necessary in the future.
